### PR TITLE
JENA-2221: Remove Apache HTTP Client usage

### PR DESCRIPTION
**Pull request by afs**
This is part 2 of a 2 change. [JENA-2221](https://issues.apache.org/jira/browse/JENA-2221).

Part 1 is #1133 [[JENA-2220](https://issues.apache.org/jira/browse/JENA-2220). This PR is based off that PR.

This part is a single commit that removes the legacy classes from jena-arq and makes Apache HttpClient artifacts scope=runtime (for jsonld-java).

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/sparql/exec/http/Service.java [hunk 1]
```diff
@@ -188,7 +188,7 @@ private static HttpClient chooseHttpClient(String serviceURL, Context context) {
             Object client = context.get(httpQueryClient);
             if ( client != null ) {
                 // Check for old HttpClient
-                if ( client instanceof org.apache.http.client.HttpClient ) {
+                if ( client.getClass().getName().equals("org.apache.http.client.HttpClient") ) {
                     LOGGER.warn("Found Apache HttpClient for content symbol "+httpQueryClient+". Jena now uses java.net.http.HttpClient");
```
Comment (rvesse) [inline jena-arq/src/main/java/org/apache/jena/sparql/exec/http/Service.java:192]: Typo `content` -> `context`

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/Async.java [hunk 2]
```diff
@@ -18,13 +18,13 @@
...
 package org.apache.jena.fuseki.ctl;
 
-import org.apache.http.HttpHeaders;
 import org.apache.jena.atlas.json.JsonBuilder;
 import org.apache.jena.atlas.json.JsonValue;
 import org.apache.jena.fuseki.async.AsyncPool;
 import org.apache.jena.fuseki.async.AsyncTask;
 import org.apache.jena.fuseki.server.DataService;
 import org.apache.jena.fuseki.servlets.HttpAction;
+import org.apache.jena.riot.web.HttpNames;
```
Comment (rvesse) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/Async.java:27]: Does `java.net.http` not have an equivalent set of constants to Apache HTTP Client?

**Review by rvesse — APPROVED** [review]

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/Async.java (hunk 2 above)
Comment (afs) [reply jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/Async.java:27]: The source code is available :smile:

If it does, it is not in the jena.net.http package which is the only module export.
Even aside from modules, the rest of the code is in `jdk.internal`.

File: jena-arq/src/main/java/org/apache/jena/sparql/exec/http/Service.java (hunk 1 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/sparql/exec/http/Service.java:192]: Fixed.

**Review by rvesse — APPROVED** [review]

**Review by kinow — APPROVED** [review]

**Pull request merged**